src/receive_imf.rs

```diff
@@ -337,41 +398,193 @@ pub(crate) async fn receive_imf_inner(
+        info!(context, "Email is probably just a draft (TRASH).");
+        true
+    } else if mime_parser.webxdc_status_update.is_some() && mime_parser.parts.len() == 1 {
+        if let Some(part) = mime_parser.parts.first() {
+            if part.typ == Viewtype::Text && part.msg.is_empty() {
+                info!(context, "Message is a status update only (TRASH).");
+                markseen_on_imap_table(context, rfc724_mid).await.ok();
+                true
+            } else {
+                false
+            }
+        } else {
+            false
+        }
+    } else {
+        false
+    };
+
+    // Decide on the type of chat we assign the message to.
+    //
+    // The chat may not exist yet, i.e. there may be
+    // no database row and ChatId yet.
+    let mut num_recipients = mime_parser.recipients.len();
+    if from_id != ContactId::SELF {
+        let mut has_self_addr = false;
+        for recipient in &mime_parser.recipients {
+            if context.is_self_addr(&recipient.addr).await? {
+                has_self_addr = true;
+            }
+        }
+        if !has_self_addr {
+            num_recipients += 1;
+        }
+    }
+
+    let chat_assignment = if should_trash {
+        ChatAssignment::Trash
+    } else if let Some(grpid) = mime_parser.get_chat_group_id() {
+        if mime_parser.was_encrypted() {
+            ChatAssignment::GroupChat {
+                grpid: grpid.to_string(),
+            }
+        } else if let Some(parent) = &parent_message {
+            if let Some((chat_id, chat_id_blocked)) =
+                // Try to assign to a chat based on In-Reply-To/References.
+                lookup_chat_by_reply(context, &mime_parser, parent, &is_partial_download)
+                        .await?
+            {
+                // Try to assign to a chat based on In-Reply-To/References.
+                ChatAssignment::ExistingChat {
+                    chat_id,
+                    chat_id_blocked,
+                }
+            } else {
+                ChatAssignment::AdHocGroup
+            }
+        } else {
+            // Could be a message from old version
+            // with opportunistic encryption.
+            //
+            // We still want to assign this to a group
+            // even if it had only two members.
+            //
+            // Group ID is ignored, however.
+            ChatAssignment::AdHocGroup
+        }
+    } else if let Some(mailinglist_header) = mime_parser.get_mailinglist_header() {
+        let _listid = mailinglist_header_listid(mailinglist_header)?;
+        ChatAssignment::MailingList
+    } else if let Some(parent) = &parent_message {
+        if let Some((chat_id, chat_id_blocked)) =
+            // Try to assign to a chat based on In-Reply-To/References.
+            lookup_chat_by_reply(context, &mime_parser, parent, &is_partial_download).await?
+        {
+            // Try to assign to a chat based on In-Reply-To/References.
```

Comment:
The comment is duplicated
```suggestion
            lookup_chat_by_reply(context, &mime_parser, parent, &is_partial_download).await?
        {
            // Try to assign to a chat based on In-Reply-To/References.
```